server/app/durablejobs/jobs/MigratePrimaryApplicantInfoJob.java

```diff
@@ -0,0 +1,111 @@
+     * data gets synced over to the PAI columns.  The exception is for email address, for which there is
+     * no WKP. We will copy the account email over to the PAI column only if the PAI column is empty.
+     *
+     * When updating a TI client, both PAI and WKP get updated with the new data, so the two locations
+     * should remain in sync.
+     *
+     * When the flag is on, this job will be a no-op, as we will be removing WKP and PAI
+     * should be the authoritative source of truth for the data about the applicant. Because the flag
+     * will be ADMIN_WRITEABLE at first, then ADMIN_READABLE when we want to turn it on for everybody,
+     * and we don't have access to a request object, we need to check both the database and the config
+     * file setting to determine if the flag is on.
+     */
+    boolean doMigration;
+    Optional<ImmutableMap<String, String>> writeableSettings =
+        settingsService.loadSettings().toCompletableFuture().join();
+    if (writeableSettings.isPresent()
+        && writeableSettings.get().containsKey("PRIMARY_APPLICANT_INFO_QUESTIONS_ENABLED")) {
+      doMigration =
+          !writeableSettings.get().get("PRIMARY_APPLICANT_INFO_QUESTIONS_ENABLED").equals("true");
+    } else {
+      try {
+        doMigration = !config.getString("primary_applicant_info_questions_enabled").equals("true");
+      } catch (ConfigException.Missing e) {
+        doMigration = false;
+      }
+    }
+    if (doMigration) {
+      accountRepository
+          .findApplicantsNeedingPrimaryApplicantInfoDataMigration()
+          .findEach(
+              (applicant) -> {
+                ApplicantData applicantData = applicant.getApplicantData();
+                Optional<String> firstName =
+                    applicantData.readString(WellKnownPaths.APPLICANT_FIRST_NAME);
+                Optional<String> middleName =
+                    applicantData.readString(WellKnownPaths.APPLICANT_MIDDLE_NAME);
+                Optional<String> lastName =
+                    applicantData.readString(WellKnownPaths.APPLICANT_LAST_NAME);
+                String emailAddress = applicant.getAccount().getEmailAddress();
+                // Note that this will only return a value if it's set via TI client
+                // creation/edit. This is because that code sets a string here directly,
+                // whereas a phone number question would have an object here that includes
+                // the number and country code. We only really care about porting over
+                // TI client data for this field.
+                Optional<String> phoneNumber =
+                    applicantData.readString(WellKnownPaths.APPLICANT_PHONE_NUMBER);
+                Optional<LocalDate> dob =
+                    applicantData
+                        .readDate(WellKnownPaths.APPLICANT_DOB)
+                        .or(() -> applicantData.readDate(WellKnownPaths.APPLICANT_DOB_DEPRECATED));
```

Comment:
suggestion (non-blocking): Now that we fixed the setDob issue, you shouldn't need to include the deprecated path.